README.md
# 海鲜鲜活下单小程序

CHANGELOG.md
## 2019-05-19 

* 1、新建项目
* 2、首页静态功能实现

## 2019-05-21

* 1、小程序名称、头像调整

## 2019-05-22

* 1、我的页面实现
* 2、商品详情页面实现

data/ajax.js
/**
 * 交互接口
 */ 

const URL = '';
// 设置DEBUG模式，使用mock虚拟数据
const DEBUG = true;
const LOADINGTEXT = '加载中...';
// 小程序拥有者身份信息
const AUTH = 'LINJINGHUI';

// 获取商品列表
let ajaxGetGoods = (pms, callback) => {
	let params = {
		page: pms.page || 1,
		size: pms.size || 10
	};
	
	uni.showLoading({title: LOADINGTEXT});
	uni.request({
		url: URL + '/goods',
		method: 'GET',
		data: params,
		header: {
			'AUTH': AUTH
		},
		dataType: 'json',
		complete: (data) => {
			uni.hideLoading();
		},
		fail: () => {
			uni.showToast({title: '网络错误，请稍后再试！', icon: 'none', position: 'bottom'});
		},
		success: (data) => {
			if (data.code === 200) {
				callback && callback(data);
			} else if (fail) {
				fail(data);
			} else {
				uni.showToast({title: '' + data.msg, icon: 'none', position: 'bottom'});
			}
		}
	});
}

// 获取商品详情
let ajaxGetGoodInfo = (pms, callback, fail) => {
	let params = {
		id: pms.id
	};
	
	uni.showLoading({title: LOADINGTEXT});
	uni.request({
		url: URL + '/goods/info',
		method: 'GET',
		data: params,
		header: {
			'AUTH': AUTH
		},
		dataType: 'json',
		complete: (data) => {
			uni.hideLoading();
		},
		fail: () => {
			uni.showToast({title: '网络错误，请稍后再试！', icon: 'none', position: 'bottom'});
		},
		success: (data) => {
			if (data.code === 200) {
				callback && callback(data);
			} else if (fail) {
				fail(data);
			} else {
				uni.showToast({title: '' + data.msg, icon: 'none', position: 'bottom'});
			}
		}
	});
}

// 获取推荐商品列表
let ajaxGetRecommendGoods = (pms, callback, fail) => {
	let params = pms;
	uni.showLoading({title: LOADINGTEXT});
	uni.request({
		url: URL + '/goods/recommend',
		method: 'GET',
		data: params,
		header: {
			'AUTH': AUTH
		},
		dataType: 'json',
		complete: (data) => {
			uni.hideLoading();
		},
		fail: () => {
			uni.showToast({title: '网络错误，请稍后再试！', icon: 'none', position: 'bottom'});
		},
		success: (data) => {
			if (data.code === 200) {
				callback && callback(data);
			} else if (fail) {
				fail(data);
			}
		}
	});
}

// 获取用户配送地址信息
let ajaxGetAddresses = (pms, callback, fail) => {
	let params = {
		userid: pms.openId
	};
	uni.showLoading({title: LOADINGTEXT});
	uni.request({
		url: URL + '/user/address',
		method: 'GET',
		data: params,
		header: {
			'AUTH': AUTH
		},
		dataType: 'json',
		complete: (data) => {
			uni.hideLoading();
		},
		fail: () => {
			uni.showToast({title: '网络错误，请稍后再试！', icon: 'none', position: 'bottom'});
		},
		success: (data) => {
			if (data.code === 200) {
				callback && callback(data);
			} else if (fail) {
				fail(data);
			}
		}
	});
}

// 保存配送地址信息, 返回id
let ajaxSaveAddresses = (pms, callback, fail) => {
	let params = {
		userid: pms.openId,
		id: pms.id || '',
		// 收货人名称
		name: pms.name,
		// 收货人电话
		mobile: pms.mobile,
		// 收货地址
		address: pms.address,
		// 收货门牌地址
		doorAddress: pms.doorAddress
	};
	if (!params.name) {
		uni.showToast({title: '请填写收货人名称', icon: 'none', position: 'bottom'});
	} else if (!params.mobile) {
		uni.showToast({title: '请填写收货人电话', icon: 'none', position: 'bottom'});
	} else if (!params.address && !params.doorAddress) {
		uni.showToast({title: '请选择收货地址', icon: 'none', position: 'bottom'});
	} else {
		uni.showLoading({title: LOADINGTEXT});
		uni.request({
			url: URL + '/user/address/save',
			method: 'POST',
			data: params,
			header: {
				'AUTH': AUTH
			},
			dataType: 'json',
			complete: (data) => {
				uni.hideLoading();
			},
			fail: () => {
				uni.showToast({title: '网络错误，请稍后再试！', icon: 'none', position: 'bottom'});
			},
			success: (data) => {
				if (data.code === 200) {
					callback && callback(data);
				} else if (fail) {
					fail(data);
				}
			}
		});
	}
}

// 下订单
let ajaxPlaceOrder = (pms, callback, fail) => {
	let goods = pms.goods || [];
	let params = {
		userid: pms.openId,
		// 收货人名称
		name: pms.name,
		// 收货人电话
		mobile: pms.mobile,
		// 收货地址
		address: pms.address,
		// 收货门牌地址
		doorAddress: pms.doorAddress,
		goods: [],
		money: pms.money || 0,
		count: pms.count || 0
	};
	for (let i = 0; i < goods.length; i++) {
		params.goods.push({
			id: goods[i].id,
			name: goods[i].name,
			count: goods[i].count,
			rprice: goods[i].rprice,
			unit: goods[i].unit
		});
	}
	if (!params.goods || params.goods.length === 0) {
		uni.showToast({title: '请先选择商品', icon: 'none', position: 'bottom'});
	} else if (!params.name) {
		uni.showToast({title: '请选择配送地址', icon: 'none', position: 'bottom'});
	} else if (!params.mobile) {
		uni.showToast({title: '请选择配送地址', icon: 'none', position: 'bottom'});
	} else if (!params.address && !params.doorAddress) {
		uni.showToast({title: '请选择配送地址', icon: 'none', position: 'bottom'});
	} else {
		console.log('==下订单==');
		console.log(params);
		uni.showLoading({title: LOADINGTEXT});
		uni.request({
			url: URL + '/user/address/save',
			method: 'POST',
			data: params,
			header: {
				'AUTH': AUTH
			},
			dataType: 'json',
			complete: (data) => {
				uni.hideLoading();
			},
			fail: () => {
				uni.showToast({title: '网络错误，请稍后再试！', icon: 'none', position: 'bottom'});
			},
			success: (data) => {
				if (data.code === 200) {
					callback && callback(data);
				} else if (fail) {
					fail(data);
				}
			}
		});
	}
}

// 获取订单列表
let ajaxGetOrders = (pms, callback) => {
	let params = {
		userid: pms.openId,
		page: pms.page || 1,
		size: pms.size || 10
	};
	
	uni.showLoading({title: LOADINGTEXT});
	uni.request({
		url: URL + '/order',
		method: 'GET',
		data: params,
		header: {
			'AUTH': AUTH
		},
		dataType: 'json',
		complete: (data) => {
			uni.hideLoading();
		},
		fail: () => {
			uni.showToast({title: '网络错误，请稍后再试！', icon: 'none', position: 'bottom'});
		},
		success: (data) => {
			if (data.code === 200) {
				callback && callback(data);
			} else if (fail) {
				fail(data);
			} else {
				uni.showToast({title: '' + data.msg, icon: 'none', position: 'bottom'});
			}
		}
	});
}

// 获取订单详情
let ajaxGetOrderInfo = (pms, callback) => {
	let params = {
		id: pms.id
	};
	
	uni.showLoading({title: LOADINGTEXT});
	uni.request({
		url: URL + '/order/info',
		method: 'GET',
		data: params,
		header: {
			'AUTH': AUTH
		},
		dataType: 'json',
		complete: (data) => {
			uni.hideLoading();
		},
		fail: () => {
			uni.showToast({title: '网络错误，请稍后再试！', icon: 'none', position: 'bottom'});
		},
		success: (data) => {
			if (data.code === 200) {
				callback && callback(data);
			} else if (fail) {
				fail(data);
			} else {
				uni.showToast({title: '' + data.msg, icon: 'none', position: 'bottom'});
			}
		}
	});
}

// 取消订单
let ajaxDelOrder = (pms, callback) => {
	let params = {
		id: pms.id
	};
	
	uni.showLoading({title: LOADINGTEXT});
	uni.request({
		url: URL + '/order/delete',
		method: 'POST',
		data: params,
		header: {
			'AUTH': AUTH
		},
		dataType: 'json',
		complete: (data) => {
			uni.hideLoading();
		},
		fail: () => {
			uni.showToast({title: '网络错误，请稍后再试！', icon: 'none', position: 'bottom'});
		},
		success: (data) => {
			if (data.code === 200) {
				callback && callback(data);
			} else if (fail) {
				fail(data);
			} else {
				uni.showToast({title: '' + data.msg, icon: 'none', position: 'bottom'});
			}
		}
	});
}

if (DEBUG) {
	const majax = require('./mock.js');
	ajaxGetGoods = majax.ajaxGetGoods;
	ajaxGetGoodInfo = majax.ajaxGetGoodInfo;
	ajaxGetRecommendGoods = majax.ajaxGetRecommendGoods;
	ajaxGetAddresses = majax.ajaxGetAddresses;
	ajaxSaveAddresses = majax.ajaxSaveAddresses;
	ajaxPlaceOrder = majax.ajaxPlaceOrder;
	ajaxGetOrders = majax.ajaxGetOrders;
	ajaxGetOrderInfo = majax.ajaxGetOrderInfo;
	ajaxDelOrder = majax.ajaxDelOrder;
}

export {ajaxGetGoods, ajaxGetGoodInfo, ajaxGetRecommendGoods, ajaxGetAddresses, ajaxSaveAddresses, ajaxPlaceOrder, ajaxGetOrders, ajaxGetOrderInfo, ajaxDelOrder};


store/index.js
import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({  
    state: {  
		// 用户信息
        user: {
			openId: '',
			nickName: '未登录',
			avatarUrl: require('../static/img/avater.png')
		},
		// 购物车
		car: {
			// 商品数量
			total: 0,
			// 商品总价
			count: 0,
			// 商品列表
			data: [],
			// 商品数量变化
			changeData: {}
		},
		// 地址
		consignees: []
    },  
	getters: {
		// 返回用户信息
		doneUser: state => {
			return state.user;
		},
		// 返回购物车中的商品信息
		doneCar: state => {
		  return state.car.data;
		},
		// 返回购物车中选中结果
		doneSelectResult: state => {
			let selectCount = 0;
			let selectMoney = 0;
			let allCount = 0;
			let selectGoods = [];
			for (let i = 0;i < state.car.data.length;i++) {
				let _item = state.car.data[i];
				allCount += _item.count;
				if (_item.select) {
					selectCount += _item.count;
					selectMoney += _item.rprice * _item.count;
					selectGoods.push(_item);
				}
			}
			return {
				allCount: allCount,
				selectCount: selectCount,
				selectMoney: selectMoney,
				selectGoods: selectGoods
			};
		}
	},
    mutations: {  
		// 设置登录用户
        setUser (state, data) {
			state.user = data;
        },
		// 商品加入购物车
		addGood (state, [good, callback]) {
			good = JSON.parse(JSON.stringify(good));
			let count = good.count;	
			// s-1 获取商品在购物车中的位置
			let index = '';
			for (let i = 0;i < state.car.data.length;i++) {
				let _item = state.car.data[i];
				if (_item.id === good.id) {
					index = i;
					good = _item;
					break;
				}
			}
						
			// s-2 更新商品数量
			if (!isNaN(count)) {
				// 有指定数量，直接设置
				good.count = count;
			} else if (index === '') {
				// 没有指定数量，且不存在购物车中
				good.count = 1;
			} else {
				// 没有指定数量，但已存在购物车中
				good.count += 1;
			}
			
			// s-2-1 判断商品库存
			if (good.count > good.stock) {
				good.count = good.stock;
				uni.showToast({title: '无法购买更多', icon: 'none', position: 'bottom'});
				return;
			}

			// s-3 维护购物车中商品
			if (index === '') {
				// 商品添加进入购物车中
				state.car.data.push(good);
			} else if (good.count <= 0) {
				// 商品从购物车中移除
				state.car.data.splice(index, 1);
			} else {
				// 更新商品
				state.car.data.splice(index, 1, good);
			}
			
			callback && callback(good);
		},
		// 设置商品勾选状态
		selectItem (state, [index, select]) {
			state.car.data[index].select = select;
		},
		// 商品全部勾选或取消勾选
		setSelectAll (state, [select]) {
			for (let i = 0; i < state.car.data.length; i++) {
				state.car.data[i].select = select;
			}
		},
		// 删除购物车中已购买成功的商品
		delGoods (state) {
			for (let i = state.car.data.length - 1;i >= 0;i--) {
				let _item = state.car.data[i];
				if (_item.select) {
					state.car.data.splice(i, 1);
				}
			}
		},
		// 设置收货地址
		setConsignee (state, data) {
			state.consignees = data || [];
		},
		// 把收货地址放到第一项
		setFirstConsignee (state, index) {
			let arr = state.consignees.splice(index, 1);
			state.consignees.unshift(arr[0]);
		},
		// 新增、编辑收货地址
		newEditConsignee (state, [data, index]) {
			if (!index && index !== 0) {
				// 新增
				state.consignees.push(data);
			} else {
				// 修改
				state.consignees.splice(index, 1, data);
			}
		}
    }  
}) 

common/global.js

// 获取微信用户信息
const getClientUser = (callback) => {
	let provider = '';
	let code = '';
	uni.getProvider({service: 'oauth'}).then((result) => { 
		result = result[1];
		provider = result.provider[0];
		return uni.login({provider: provider});
	}).then((result) => { 
		result = result[1];
		code = result.code;
		return uni.getUserInfo({provider: provider});
	}).then((result) => { 
		result = result[1];
		let userInfo = result.userInfo;
		userInfo.provider = provider;
		userInfo.code = code;
		userInfo.signature = result.signature;
		callback &&　callback(userInfo);
	}).catch(err => {
		uni.showModal({ title: '获取用户信息失败', content: '接口调用失败', showCancel: false });
	});
}

// 页面跳转
const turnPage = (type, obj) => {
	let url = '';
	switch (type) {
		case 'home':
			url = '/pages/home/main';
			break;
		case 'my':
			url = '/pages/my/main';
			break;
		case 'car':
			url = '/pages/car/main';
			break;
		case 'detail':
			url = '/pages/detail/main?id=' + obj.id;
			break;
		case 'consignee':
			url = '/pages/consignee/main?from=' + obj;
			break;
		case 'csg-creat':
			url = '/pages/consignee/creat';
			if (obj || obj === 0) {
				url += '?index=' +　obj;
			}
			break;
		case 'order':
			url = '/pages/order/main';
			break;
		case 'order-info':
			url = '/pages/order/detail?id=' + obj;
			break;
		default:
			url = '/pages/home/main';
			break;
	}
	uni.navigateTo({url: url});
}

// // 计算折扣后的真实价格
// const countRealPrice = (good) => {
// 		return (good.price * good.rebate / 10).toFixed(2);
// }

export {getClientUser, turnPage};

store/index.1.js
import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({  
    state: {  
        user: {
			openId: '',
			nickName: '',
			avatarUrl: require('../static/img/avater.png')
		},
		car: {
			total: 0,
			count: 0,
			data: [],
			// 商品数量变化
			changeData: {}
		}
    },  
	getters: {
		doneCardata: state => {
		  return state.car.data;
		}
	},
    mutations: {  
        setUser (state, data) {
			state.user = data;
        },
		// 商品加入购物车
		addGoodToCar (state, good) {
			let count = good.count;	
			// 判断商品是否已经在购物车中，如果是则更新数量，不是则添加进购物车 ，如果数量是0 则移除商品
			// s-1 获取商品在购物车中的位置
			let index = '';
			for (let i = 0;i < state.car.data.length;i++) {
				let _item = state.car.data[i];
				if (_item.id === good.id) {
					index = i;
					good = _item;
					break;
				}
			}
			// s-2 更新商品数量
			if (good.from === 'detail') {
				// 商品来源详情页
				good.count += 1;
			} else {
				good.count = count;	
			}
			// if (count === '+1') {
			// 	(good.count === undefined) && (good.count = 0);
			// 	good.count += 1;
			// } else {
			// 	
			// }
			// s-2-1 维护商品数量变化
			state.car.changeData = JSON.parse('{\"' + good.id + '\":' + good.count + '}');
			// s-3 判断数量是0，移除商品
			if (index !== '' && count <= 0) {
				state.car.data.splice(index, 1);
			} else if (index === '') {
				// s-4 商品不在购物车中，添加进购物车
				state.car.data.push(good);
			}
			// s-5 统计购物车中的商品总数，总价
			let total = 0;
			let totalCount = 0;
			for (let z = 0;z < state.car.data.length;z++) {
				let _good = state.car.data[z];
				let _count = _good.count;
				totalCount += _count;
				total += _good.rprice * _count;
			}
			state.car.count = totalCount;
			state.car.total = total.toFixed(2);
			console.log('==store==');
			console.log(state.car.data);
		}
    }  
}) 

data/mock.js
import Mock from '@/static/mock/mock.js';
const Random = Mock.Random;
const dure = 1000;
const LOADINGTEXT = '加载中...';

// 获取商品列表
export function ajaxGetGoods (pms, callback) {
	let data = Mock.mock({
		'msg': '',
		'code': 200,
		// 总记录数
		'total': 32,
		'result|6-10': [{
			'id': '@id()',
			// 图片地址
			'pic': "@image(80x80)",
			// 商品名称
			'name|1': '@ctitle(3, 8)',
			// 商品描述
			'desc': '@ctitle(10, 50)',
			// 购买单位
			'unit|1': ['1kg-1.5kg/份', '2-2.5g/个', '3.5元/斤'],
			// 单价
			'price|1': [3.5, 5, 11, 12.5, 22.3],
			// 折扣
			'rebate|1': [9.5, 8.5, 10, 5, 4.5],
			// 真实单价|折后单价
			'rprice|1': [3.5, 5, 11, 12.5, 22.3],
			// 库存
			'stock|1': [10, 5, '', 50],
			// 标签
			'label|0-3': [
				{ 
					'text|1': ['折扣', '清仓', '极鲜'], 
					'bcolor|1': ['#bf2058', '#ced642', '#fbb519'], 
					'color': '#000'
				}
			]
			// 购买数量
			// 'count': 1
			// 实际发货数量
			// 'realCount': 1
		}]
	});
	uni.showLoading({title: LOADINGTEXT});
	setTimeout(function () {uni.hideLoading(); callback && callback(data);}, Math.floor(Math.random() * dure));
}

// 获取商品详情
export function ajaxGetGoodInfo (pms, callback) {
	let data = Mock.mock({
		'msg': '',
		'code': 200,
		'result': {
			'id': pms.id,
			'pic': "@image(80x80)",
			// 广告图片
			'covers|1-2': ['https://img-cdn-qiniu.dcloud.net.cn/uniapp/images/shuijiao.jpg', 'https://img-cdn-qiniu.dcloud.net.cn/uniapp/images/muwu.jpg', 'https://img-cdn-qiniu.dcloud.net.cn/uniapp/images/cbd.jpg'],
			// 商品名称
			'name|1': '@ctitle(3, 8)',
			// 商品描述
			'desc': '@ctitle(10, 50)',
			// 购买单位
			'unit|1': ['1kg-1.5kg/份', '2-2.5g/个', '3.5元/斤'],
			// 单价
			'price|1': [3.5, 5, 11, 12.5, 22.3],
			// 折扣
			'rebate|1': [9.5, 8.5, 10, 5, 4.5],
			// 真实单价|折后单价
			'rprice|1': [3.5, 5, 11, 12.5, 22.3],
			// 库存
			'stock|1': [10, 5, '', 50],
			// 标签
			'label|0-3': [
				{ 
					'text|1': ['折扣', '清仓', '极鲜'], 
					'bcolor|1': ['#bf2058', '#ced642', '#fbb519'], 
					'color': '#000'
				}
			],
			// 产地
			'pplace|1': ['福建', '连江', '苔箓', '北郊'],
			// 商品详情图片
			'dtlpics|1-5': ["@image(200x280)"],
		}
	});
	uni.showLoading({title: LOADINGTEXT});
	setTimeout(function () {uni.hideLoading(); callback && callback(data);}, Math.floor(Math.random() * dure));
}

// 获取推荐商品列表
export function ajaxGetRecommendGoods (pms, callback) {
	let data = Mock.mock({
		'msg': '',
		'code': 200,
		'result|1-10': [{
			'id': '@id()',
			// 图片地址
			'pic': "@image(80x80)",
			// 商品名称
			'name|1': '@ctitle(3, 8)',
			// 商品描述
			'desc': '@ctitle(10, 50)',
			// 购买单位
			'unit|1': ['1kg-1.5kg/份', '2-2.5g/个', '3.5元/斤'],
			// 单价
			'price|1': [3.5, 5, 11, 12.5, 22.3],
			// 折扣
			'rebate|1': [9.5, 8.5, 10, 5, 4.5],
			// 真实单价|折后单价
			'rprice|1': [3.5, 5, 11, 12.5, 22.3],
			// 库存
			'stock|1': [10, 5, '', 50],
			// 标签, 最多3个
			'label|0-3': [
				{ 
					'text|1': ['折扣', '清仓', '极鲜'], 
					'bcolor|1': ['#bf2058', '#ced642', '#fbb519'], 
					'color': '#000'
				}
			]
		}]
	});
	uni.showLoading({title: LOADINGTEXT});
	setTimeout(function () {uni.hideLoading(); callback && callback(data);}, Math.floor(Math.random() * dure));
}

// 获取用户配送地址信息
export function ajaxGetAddresses (pms, callback) {
	let data = Mock.mock({
		'msg': '',
		'code': 200,
		'result|0-3': [{
			'id': '@id()',
			// 收获人名称
			'name': '@ctitle(3, 8)',
			// 收获人电话
			'mobile': /^1[385][1-9]\d{8}/,
			// 收获地址
			'address': '@ctitle(10, 20)',
			// 收获门牌地址
			'doorAddress': '@ctitle(5, 10)'
		}]
	});
	uni.showLoading({title: LOADINGTEXT});
	setTimeout(function () {uni.hideLoading(); callback && callback(data);}, Math.floor(Math.random() * dure));
}

// 保存配送地址信息, 返回id
export function ajaxSaveAddresses (params, callback) {
	if (!params.name) {
		uni.showToast({title: '请填写收货人名称', icon: 'none', position: 'bottom'});
	} else if (!params.mobile) {
		uni.showToast({title: '请填写收货人电话', icon: 'none', position: 'bottom'});
	} else if (!params.address && !params.doorAddress) {
		uni.showToast({title: '请选择收货地址', icon: 'none', position: 'bottom'});
	} else {
		let data = Mock.mock({
			'msg': '',
			'code': 200,
			'result': 'adrressId'
		});
		uni.showLoading({title: LOADINGTEXT});
		setTimeout(function () {uni.hideLoading(); callback && callback(data);}, Math.floor(Math.random() * dure));
	}
}

// 下订单
export function ajaxPlaceOrder (pms, callback) {
	let goods = pms.goods || [];
	let params = {
		userid: pms.openId,
		// 收货人名称
		name: pms.name,
		// 收货人电话
		mobile: pms.mobile,
		// 收货地址
		address: pms.address,
		// 收货门牌地址
		doorAddress: pms.doorAddress,
		goods: [],
		money: pms.money || 0,
		count: pms.count || 0
	};
	for (let i = 0; i < goods.length; i++) {
		params.goods.push({
			id: goods[i].id,
			name: goods[i].name,
			count: goods[i].count,
			rprice: goods[i].rprice,
			unit: goods[i].unit
		});
	}
	if (!params.goods || params.goods.length === 0) {
		uni.showToast({title: '请先选择商品', icon: 'none', position: 'bottom'});
	} else if (!params.name) {
		uni.showToast({title: '请选择收货地址', icon: 'none', position: 'bottom'});
	} else if (!params.mobile) {
		uni.showToast({title: '请选择收货地址', icon: 'none', position: 'bottom'});
	} else if (!params.address && !params.doorAddress) {
		uni.showToast({title: '请选择收货地址', icon: 'none', position: 'bottom'});
	} else {
		console.log('==下订单==');
		console.log(params);
		let data = Mock.mock({
			'msg': '',
			'code': 200
		});
		uni.showLoading({title: LOADINGTEXT});
		setTimeout(function () {uni.hideLoading(); callback && callback(data);}, Math.floor(Math.random() * dure));
	}
}

// 获取订单列表
export function ajaxGetOrders (pms, callback) {
	let data = Mock.mock({
		'msg': '',
		'code': 200,
		// 总记录数
		'total': 21,
		'result|10': [
			{
				// 订单ID
				'id': '@id()',
				// 商品列表
				'goods|2': [{
					'id': '@id()',
					'pic': "@image(80x80)",
					'name': '@ctitle(3, 8)',
					'count|1': [1, 2, 3]
				}],
				// 订单状态 1：待接单，2：备货中，3：配送中，4：已完成
				'status|1': [1, 2, 3, 4],
				// 总数量
				'count': 10,
				// 金额
				'money': 100,
				'time': '@date()'
			}
			
		]
	});
	uni.showLoading({title: LOADINGTEXT});
	setTimeout(function () {uni.hideLoading(); callback && callback(data);}, Math.floor(Math.random() * dure));
}

// 获取订单详情
export function ajaxGetOrderInfo (pms, callback) {
	let data = Mock.mock({
		'msg': '',
		'code': 200,
		'result': {
			// 订单ID
			'id': '@id()',
			// 商品列表
			'goods|1-5': [{
				'id': '@id()',
				'pic': "@image(80x80)",
				'name': '@ctitle(3, 8)',
				'count|1': [1, 2, 3],
				// 订购价格|折后价格
				'rprice|1': [3.5, 5, 11, 12.5, 22.3],
				// 最新价格
				'nprice|1': ['', 5, 11, 12.5, 22.3]
			}],
			// 订单状态 1：待接单，2：备货中，3：配送中，4：已完成
			'status|1': [1, 2, 3, 4],
			// 总数量
			'count': 10,
			// 金额
			'money': 100,
			// 快递费用
			'expMoney': 10,
			'time': '@date()',
			// 收货人名称
			'name': '@name()',
			// 收货人电话
			'mobile': /^1[385][1-9]\d{8}/,
			// 收货地址
			'address': '@ctitle(10, 20)',
			// 收货门牌地址
			'doorAddress': '@ctitle(5, 10)'
		}
	});
	uni.showLoading({title: LOADINGTEXT});
	setTimeout(function () {uni.hideLoading(); callback && callback(data);}, Math.floor(Math.random() * dure));
}

// 取消订单
export function ajaxDelOrder (pms, callback) {
	let data = Mock.mock({
		'msg': '',
		'code': 200,
		'result': {}
	});
	uni.showLoading({title: LOADINGTEXT});
	setTimeout(function () {uni.hideLoading(); callback && callback(data);}, Math.floor(Math.random() * dure));
}